### PR TITLE
Update travis to use builder.py

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,9 +8,15 @@ osx_image:
   - xcode10.1
 compiler: clang
 
+addons:
+  homebrew:
+    update: true
+    packages:
```
Comment (justinboswell) [inline .travis.yml]: so do we no longer install cmake? Is there a system one to use?

File: codebuild/builder.py [hunk 2]
```diff
@@ -143,6 +148,16 @@ def __repr__(self):
             "-DENABLE_SANITIZERS=ON",
         ],
     },
+    'macos': {
+        'architectures': {
+            'x86': {
```
Comment (justinboswell) [inline codebuild/builder.py]: Wait, we support 32-bit OSX? Is that even a thing at this point?

File: .travis.yml (hunk 1 above)
Comment (ColdenCullen) [reply .travis.yml]: I mean, I guess? I didn't remove anything, and the builds passed, so praise the travis gods?

File: codebuild/builder.py (hunk 2 above)
Comment (ColdenCullen) [reply codebuild/builder.py]: Hell if I know, but couldn't see a reason not to put it in there

File: .travis.yml (hunk 1 above)
Comment (justinboswell) [reply .travis.yml]: :muscle:

File: codebuild/builder.py (hunk 2 above)
Comment (justinboswell) [reply codebuild/builder.py]: won't this double our travis builds?

Comment (ColdenCullen) [reply codebuild/builder.py]: No, because travis builds aren't defined by this script, they're defined by travis.yml.

**Pull request merged**